### PR TITLE
[br-tests] add more steps to test_advertising_proxy.py

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Add steps which verify the bugfix contributed to OTBR: https://github.com/openthread/ot-br-posix/pull/1020
That is, verify that SRP updates which modify both a host and a service can be successfully completed.

Depends-On: openthread/ot-br-posix#1020

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

File: tests/scripts/thread-cert/border_router/test_advertising_proxy.py [hunk 1]
```diff
@@ -176,7 +176,34 @@ def test(self):
...
+        #
+        # 8. Update both the host and the service in a loop and make sure the
+        #    Advertising Proxy can follow.
+        #
+
+        for host_address_suffix, service_port in ((1, 12341), (2, 12342), (3, 12343), (2, 12345)):
+            host_address = f'2001::{host_address_suffix}'
+            client.srp_client_set_host_address(host_address)
+            client.srp_client_clear_service('my-service', '_ipps._tcp')
+            client.srp_client_add_service('my-service', '_ipps._tcp', service_port)
```
Comment (wgtdkp) [inline tests/scripts/thread-cert/border_router/test_advertising_proxy.py:199]: If I understand it correctly, this test case is trying to include `Host` and `Service` updates in a single SRP message. I know that the SRP client will wait for `OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_DELAY` (10ms) before sending a SRP message when `Host` or `Service` info is updated to collect multiple updates in a single SRP message. Considering that the commands in the test script will be mapped to separate CLI commands, there is no guarantee that `client.srp_client_set_host_address` and `client.srp_client_add_service` can be executed in `10ms`.

Suggest changing to
```c++
client.srp_client_clear_service('my-service', '_ipps._tcp')
client.srp_client_set_host_address(host_address)
client.srp_client_add_service('my-service', '_ipps._tcp', service_port)
```

Comment (Damian-Nordic) [reply tests/scripts/thread-cert/border_router/test_advertising_proxy.py:199]: I think it's a valid point. I'll change it as you suggested.

**Comment by LuDuda** [comment]
Can we re-run the CI? If I'm not mistaken the error seems to be unrelated to this change.

**Review by wgtdkp — APPROVED** [review]
LGTM 👍
Thanks!

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**